### PR TITLE
[jcof] add types for jcof@1.0

### PR DESCRIPTION
**Pull request by MatyiFKBT**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/jcof/jcof-tests.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import jcof = require('jcof');
+
+const exampleObject = {
+	people: [
```
Comment (DangerBotOSS) [inline types/jcof/jcof-tests.ts:4]: - :no_entry_sign: The root .editorconfig style specifies spaces for whitespace. Please use spaces in new or changed types.

File: types/jcof/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,23 @@
+{
+	"compilerOptions": {
```
Comment (DangerBotOSS) [inline types/jcof/tsconfig.json:2]: - :no_entry_sign: The root .editorconfig style specifies spaces for whitespace. Please use spaces in new or changed types.

File: types/jcof/tslint.json [hunk 3]
```diff
@@ -0,0 +1,3 @@
+{
+	"extends": "@definitelytyped/dtslint/dt.json"
```
Comment (DangerBotOSS) [inline types/jcof/tslint.json:2]: - :no_entry_sign: The root .editorconfig style specifies spaces for whitespace. Please use spaces in new or changed types.

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## jcof ([<kbd>unpkg</kbd>](https://unpkg.com/browse/jcof@latest/))
was missing the following properties:
1. ParseError
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 83a398a3926d1646e18de8126eb356ccfc9f9722
</p>

**Comment by jakebailey** [comment]
Based on the above comment and skimming the code, ParseError appears to be an exported part of this package's API; you might want to add it.

**Review by jakebailey — APPROVED** [review]

**Comment by MatyiFKBT** [comment]
Ready to merge

**Pull request merged**